elastic/kibana
# [Uptime] get last successful check instead of last successful step

## Summary

Fixes https://github.com/elastic/kibana/issues/117228

Changes last successful **step** query to last successful **check** query for displaying in the check details page step accordion. 

<img width="811" alt="Screen Shot 2021-11-16 at 5 30 08 PM" src="https://user-images.githubusercontent.com/11356435/142335122-e773e8c8-5469-4760-8269-0ceed5632faf.png">

**Why is this important?**

The previous query did not account for ensuring that `heartbeat/summary` docs were available for the associated step found with the `last_successful_step` query. The absence of heartbeat/summary docs indicates that the check is in progress, or has crashed entirely. For this reason, we should only use data from check groups that have associated `heartbeat/summary` documents.  Indeed, we only show checks with `heartbeat/summary` documents on the overview page and so we only want to use fully completed checks with `heartbeat/summary` docs elsewhere in the app.

Previously, we checked for the last successful **step**. This did not take into account whether or not the whole journey was successful, but only looked for the last successful step, which could be present in a failing journey. From product, the decision was to move forward with looking for the last successful **check**, and taking the screenshot from matching step for that check.


### Testing

#### Testing for monitors with locations
- Connect to edge data via `oblt-cli`
- Navigate to Uptime overview page, and filter for browser monitors.
- Click on `My monitor - inline`
- My monitor inline frequently has failures, search through the data to find a failure.
- Click on the failing check, find the failing step, and click on the accordion to view the failure screenshot, as well as the last successful screenshot.
- Ensure that the last successful screenshot is present, and that the last successful screenshot is the screenshot that is expected. The screenshot should be the screenshot for that step from the last successful check.

#### Testing for monitors without locations
- Run es locally via `yarn es snapshot`. Run your own heartbeat monitors, ensuring that at least one synthetics monitor is successful and connect local Kibana
- Stop local heartbeat, edit your synthetic monitor to cause it to fail, and restart heartbeat
- Navigate to Uptime overview page, and filter for browser monitors.
- Click on the monitor you wish to test and find the latest failing check
- Click on the failing check, find the failing step, and click on the accordion to view the failure screenshot, as well as the last successful screenshot.
- Ensure that the last successful screenshot is present, and that the last successful screenshot is the screenshot that is expected. The screenshot should be the screenshot for that step from the last successful check.
